card/mod/01_core/set/all/templating.rb

```diff
@@ -12,29 +12,34 @@ def template
   # note that a *default template is never returned for an existing card.
   @template ||= begin
     @virtual = false
+
+    # NEW CARDS
     if new_card?
-      default_card = rule_card :default, :skip_modules=>true
+      default = rule_card :default, :skip_modules=>true
 
       dup_card = dup
-      dup_card.type_id = default_card ? default_card.type_id : Card.default_type_id
-
+      dup_card.type_id = default ? default.type_id : Card.default_type_id
 
-      if content_card = dup_card.structure_rule_card
+      if structure = dup_card.structure_rule_card
         @virtual = true if junction?
-        content_card
+        self.type_id = structure.type_id if assign_type_to?(structure)
```

Comment:
You know I'm not against slipping a small fix in with some other changes :)

I know it was about virtual cards, but only when they are real? or !real?   Is it just about the type_id on the card object (in memory), so mainly the !real? case?  Since nests and URLs probably don't effect the type when it is a real card ...